src/js/points.js

```diff
@@ -125,22 +125,28 @@ function parseCityLabelItem(item) {
 }
 
 function loadCityLabels() {
-  loadData(
-    new URL("../../asset/labels.tsv", import.meta.url),
-    parseCityLabelItem,
-    cityLabels, // Store city labels in the cityLabels array
-    () => {},
-  );
+  return new Promise((resolve) => {
+    loadData(
+      new URL("../../asset/labels.tsv", import.meta.url),
+      parseCityLabelItem, // sets cityLabelsByClusterId
+      cityLabels, // Store city labels in the cityLabels array
+      () => {
+        resolve();
```

Comment:
This callback is triggered once `loadData` processes all the TSV items. By wrapping the entire thing in a `Promise` and resolving it here, we ensure that all the data is processed when the `Promise` resolves.